iter.go

```diff
@@ -103,49 +106,25 @@ func (i *LogIter) ValueLen() uint64 {
 // Key returns the full key at the current position.
 // This method will return a result only once per iteration.
 func (i *LogIter) Key() ([]byte, error) {
-	return i.KeyChunk(-1)
+	return ioutil.ReadAll(i.KeyReader())
 }
 
-// KeyChunk returns a chunk of the key at the current position.
-func (i *LogIter) KeyChunk(maxlen int) ([]byte, error) {
-	var max, size C.uint64_t
-	var ptr *C.uint8_t
-
-	if maxlen < 0 {
-		max = C.sparkey_logiter_keylen(i.iter)
-	} else {
-		max = C.uint64_t(maxlen)
-	}
-
-	rc := C.sparkey_logiter_keychunk(i.iter, i.log, max, &ptr, &size)
-	if rc != rc_SUCCESS {
-		return nil, Error(rc)
-	}
-	return C.GoBytes(unsafe.Pointer(ptr), C.int(size)), nil
+// KeyReader returns an io.Reader for the key. The reader also implements
+// io.WriterTo. The reader is no longer valid once the iterator has proceeded.
+func (i *LogIter) KeyReader() io.Reader {
```

Comment:
I think we need to declare our own Reader interface, as we are exposing not just `Read` but also `WriteTo`